python/cross_service/resilient_service/load_balancer.py

```diff
@@ -0,0 +1,176 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+import time
+
+import boto3
+from botocore.exceptions import ClientError
+
+log = logging.getLogger(__name__)
+
+
+class LoadBalancerError(Exception):
+    pass
+
+# snippet-start:[python.example_code.workflow.ResilientService_LoadBalancer]
+class LoadBalancer:
+    """Encapsulates Elastic Load Balancer actions."""
+    def __init__(
+            self, target_group_name, load_balancer_name, elb_client):
+        """
+        :param target_group_name: The name of the target group associated with the load balancer.
+        :param load_balancer_name: The name of the load balancer.
+        :param elb_client: A Boto3 Elastic Load Balancer client.
+        """
+        self.target_group_name = target_group_name
+        self.load_balancer_name = load_balancer_name
+        self.elb_client = elb_client
+        self._endpoint = None
+
+    @classmethod
+    def from_client(cls, resource_prefix):
+        """
+        Creates this class from a Boto3 client.
+
+        :param resource_prefix: The prefix to give to AWS resources created by this class.
+        """
+        elb_client = boto3.client('elbv2')
+        return cls(
+            f"{resource_prefix}-tg",
+            f"{resource_prefix}-lb",
+            elb_client)
+
+    def endpoint(self):
+        """
+        Gets the HTTP endpoint of the load balancer.
+
+        :return: The endpoint.
+        """
+        if self._endpoint is None:
+            try:
+                response = self.elb_client.describe_load_balancers(Names=[self.load_balancer_name])
+                self._endpoint = response['LoadBalancers'][0]['DNSName']
+            except ClientError as err:
+                raise LoadBalancerError(
+                    f"Couldn't get the endpoint for load balancer {self.load_balancer_name}: {err}")
+        return self._endpoint
+
+    def create_target_group(self, protocol, port, vpc_id):
+        """
+        Creates an Elastic Load Balancer target group. The target group specifies how
+        the load balancer forward requests to instances in the group and how instance
+        health is checked.
+
+        :param protocol: The protocol to use to forward requests, such as 'HTTP'.
+        :param port: The port to use to forward requests, such as 80.
+        :param vpc_id: The ID of the VPC in which the load balancer exists.
+        :return: Data about the newly created target group.
+        """
+        try:
+            response = self.elb_client.create_target_group(
+                Name=self.target_group_name, Protocol=protocol, Port=port,
+                HealthCheckPath='/healthcheck', HealthCheckIntervalSeconds=10,
```

Comment:
Is it explained anywhere why you chose these particular health check settings?
For example, HealthyThresholdCount.

Reply:
Good call, I added info the comment.